src/components/Contact/contactform.tsx

```diff
@@ -72,10 +78,15 @@ const ContactForm = ({ contact, image, setIsEditing }: ContactInfoProps) => {
   // TODO: Submit ContactForm to database
   const submitForm = (e: React.FormEvent) => {
     e.preventDefault()
-    // TODO: submit to firestore here
-    setIsEditing(false)
+    if (setIsEditing !== undefined) setIsEditing(false)
 
-    // TODO: reload page here
+    if (isNewContact) {
+      // TODO: generate UUID for the new contact (or let firestore do it)
+      // TODO: submit to firestore here
+    } else {
+      // TODO: update firestore entry for existing uuid
+    }
+    // TODO: go to /contact/[UUID]
```

Comment:
we might just use the server timestamp as the uid for the contacts

Reply:
No problem, can tackle this in #81